apps/giphy/giphy.star

```diff
@@ -0,0 +1,132 @@
+"""
+Applet: Giphy
+Summary: Displays Giphy gifs
+Description: Displays a random gif based on a search query. Powered by Giphy.com.
+Author: Ricky Smith (DigitallyBorn)
+"""
+
+load("cache.star", "cache")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("random.star", "random")
+load("render.star", "render")
+load("schema.star", "schema")
+load("secret.star", "secret")
+
+SEARCH_URL = "https://api.giphy.com/v1/gifs/search?api_key={}&q={}&limit=25&offset=0&rating={}&lang=en"
+API_TOKEN_ENCRYPTED = """
+AV6+xWcEVvNa2fJsV0ttCcP9TohstcHBVAANlAJL8RgTvlMp1nuHmuSOsR7wFL/QbBO9Sts9ryKe
+dE484RsAxjJyO+H4l9ft1kA4HLAD7dn8Uh1ORo2hYHccpcTYWILEUlCAdUuyI0eQ5Iydq3kacvu7
+NtIzvRSGyZyUxtx6i+TsQdsr8ES/b+i2DBkItCNhnl5hLa87
+"""
+
+GIPHY_LOGO = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAABlElEQVR4nMWXMU/CQBTHf70YExBi
+mDoRDYsrA2F3YOQbMLBQdw2fwuAOLiz4BdjgGxhMjJsLO4NaU8LAUoe7SimlLbWcv+TS9u7d+99r
+cu/eGSQjDzSBBlADKkBRjTnAHJgBU2AMrBL63YsJ9JRzN2Fz1BwzrWgHsA8QDDZb+UjMKTD6g2Cw
+jZTPWNFJhqJem8SJZxlpWOS/GL73DjAIrsR1WntX+fzyQf2pDEBpOMRer6OCArCARwChOkzgPm5W
+UPT17fOQKSgNEzYR94DbUFO3L5/GTfiwZW19J4j8AbgTyORgRVlmjAXkBTIjFbLy+tVu7/yFAAWg
+KZBpUDeNE2TuTY0x2N4IrYs+5VyVkA3ipyaQCT8zyrkql2f1OLOKYHPKZEICUYCiiLc5DgJ5jOnG
+EchDXDdzgawcdDMTyHJFN1OBrJGWGkWXwFggC7PI3Z4xA2DlnU4m8A6c75jN3HTur0vg2MHeb+AK
+WHj7eAF00ykcRFdp7aCt9Anyb8WeJ669vPWjvaD3c7QrjBE16CPzS9sPWXwFXHChq40AAAAASUVO
+RK5CYII=
+""")
+
+def main(config):
+    API_TOKEN = secret.decrypt(API_TOKEN_ENCRYPTED) or config.get("dev_api_key")
+    search_query = config.str("search_query", "do it")
+    rating = config.str("rating", "g")
+
+    results = search(search_query, rating, API_TOKEN)
+    image_count = len(results["data"])
+
+    image_data = None
+    for _ in range(25):
+        image = results["data"][random.number(0, image_count - 1)]
```

Comment:
This will error out when `image_count == 0`.

Reply:
Should be handled now.